x-pack/plugins/security_solution/server/endpoint/routes/actions/response_actions.ts

```diff
@@ -313,7 +313,7 @@ function responseActionRequestHandler<T extends EndpointActionDataParameterTypes
     const casesClient = await endpointContext.service.getCasesClient(req);
     const connectorActions = (await context.actions).getActionsClient();
     const responseActionsClient: ResponseActionsClient = getResponseActionsClient(
-      req.body.agent_type || 'endpoint',
+      req.body.agent_type,
```

Comment:
Is `|| 'endpoint'` needed? the value for `agent_type` is always set - defaults to `endpoint`.  Or did you try to make an API request with `?agent_type=` and that failed?

Just trying to understand the issue was here 😄

Reply:
I am having issue with this default value, because:
- it's supposed to be set in the api schema 
```
  agent_type: schema.oneOf(
    // @ts-expect-error TS2769: No overload matches this call
    RESPONSE_ACTION_AGENT_TYPE.map((agentType) => schema.literal(agentType)),
    { defaultValue: 'endpoint' }
  ),
  ```
but it does not apparently because the tests say so ;/ 

I can't understand why that happens, but for the safety reasons I think I would leave 
```
      req.body.agent_type || 'endpoint',
```
still here. 
What do you think?

Reply:
I think that our jest tests are just skipping the schema validation totally. I got confused because also my cypress tests failed heavily - but that might have been a different issue on CI (maybe?)

Reply:
I think I will have to adjust the callRoute or the requestMock to validate request body according to schema. But that will be out of this scope and will probably break tons of tests.

Reply:
Thanks for clarifying. The changed did not have any negative side effects because we are assuming `agent_type` default to be `endpoint`, but just seemed unnecessary. I'm fine with you leaving it in place - I just wanted to understand what lead to it. 

The tests you probably saw failing are likely for the route handler - which you are correct: do not invoke the full core `Router` flow but rather are designed to (unit) test only the handler - thus it does not flow through the schema.